### PR TITLE
Refactor basystem

### PR DESCRIPTION
**Pull request by aleitner**
Stop using streams

### CODE REVIEW

File: pkg/bwagreement/dbx/gen.go [hunk 1]
```diff
@@ -5,3 +5,5 @@ package bwagreement
 
 //go:generate dbx.v1 golang -d postgres bwagreement.dbx .
 //go:generate dbx.v1 schema -d postgres bwagreement.dbx .
+//go:generate dbx.v1 golang -d sqlite3 bwagreement.dbx .
+//go:generate dbx.v1 schema -d sqlite3 bwagreement.dbx .
```
Comment (egonelbre) [inline pkg/bwagreement/dbx/gen.go]: you can specify these on the same line

File: pkg/bwagreement/server.go [hunk 2]
```diff
@@ -110,75 +107,66 @@ func (dbm *DBManager) GetBandwidthAllocations(ctx context.Context) (rows []*dbx.
...
+
+	return reply, nil
 }
 
 func (s *Server) verifySignature(ctx context.Context, ba *pb.RenterBandwidthAllocation) error {
 	// TODO(security): detect replay attacks
-	pi, err := provider.PeerIdentityFromContext(ctx)
+
+	//Deserealize RenterBandwidthAllocation.GetData() so we can get public key
+	rbad := &pb.RenterBandwidthAllocation_Data{}
```
Comment (egonelbre) [inline pkg/bwagreement/server.go]: Is there a reason we are not using the pub-key that is passed along with TLS connection?

Comment (aleitner) [reply pkg/bwagreement/server.go]: We need the uplink's public key. These messages are being sent from the Storage nodes. The storage nodes need to hold onto the public key used for each allocation

File: pkg/bwagreement/config.go [hunk 3]
```diff
@@ -20,18 +23,24 @@ var (
...
 
-	ns, err := NewServer(c.DatabaseDriver, c.DatabaseURL, zap.L(), k)
+	zap.S().Debug("Starting Bandwidth Agreement Receiver...")
+
+	u, err := url.Parse(c.DatabaseURL)
+	if err != nil {
+		return errs.New("Invalid Database URL: %+v", err)
+	}
+
+	ns, err := NewServer(u.Scheme, u.Path, zap.L(), k)
```
[the remark this answers is not in the document]
Comment (aleitner) [reply pkg/bwagreement/config.go]: done

**Pull request merged**